### PR TITLE
documentation: update framework version links

### LINKED ISSUE #4143: Broken links for framework versions
**What did you find confusing? Please describe.**
The links for "framework_version" parameter in estimators point to the readme in this repo, which then link back to the documentation.  It's very hard to find valid versions, and dependencies.
Examples:
* Tensorflow: https://github.com/aws/sagemaker-python-sdk/blob/16d1556e24ef15e03d0ec779f3420df00b8cbad3/src/sagemaker/tensorflow/estimator.py#L62C47-L62C47
* Sklearn: https://github.com/aws/sagemaker-python-sdk/blob/16d1556e24ef15e03d0ec779f3420df00b8cbad3/src/sagemaker/sklearn/estimator.py#L78

**Describe how documentation can be improved**
Fix the documentation to list version, or fix links to point to an actual version list.

**Additional context**
Driving me crazy

### PR DESCRIPTION
**Pull request by trungleduc**
*Issue #, if available:*
- closes #4143 

*Description of changes:*
- Update broken links

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by goelakash — APPROVED** [review]

File: src/sagemaker/huggingface/model.py [hunk 1]
```diff
@@ -138,12 +138,10 @@ def __init__(
...
             tensorflow_version (str): TensorFlow version you want to use for
                 executing your inference code. Defaults to ``None``. Required unless
-                ``pytorch_version`` is provided. List of supported versions:
-                https://github.com/aws/sagemaker-python-sdk#huggingface-sagemaker-estimators.
+                ``pytorch_version`` is provided. The current supported version is ``2.4.1``.
             pytorch_version (str): PyTorch version you want to use for
                 executing your inference code. Defaults to ``None``. Required unless
-                ``tensorflow_version`` is provided. List of supported versions:
-                https://github.com/aws/sagemaker-python-sdk#huggingface-sagemaker-estimators.
+                ``tensorflow_version`` is provided. The current supported versions are ``1.7.1`` and ``1.6.0``.
```
Comment (knikure) [inline src/sagemaker/huggingface/model.py:144]: we will need to swap this. I this mixed tensorflow_version with pytorch_version

**Review by knikure — CHANGES_REQUESTED** [review]
Just a minor comment. Rest LGTM!

File: src/sagemaker/huggingface/model.py (hunk 1 above)
Comment (trungleduc) [reply src/sagemaker/huggingface/model.py:144]: Not sure I understand the issue, I get this reference from here

https://github.com/aws/sagemaker-python-sdk/blob/488adba580378892cfa138efd578bfc7aad2e856/src/sagemaker/huggingface/processing.py#L69-L74

Do we need to swap the version number?

**Review by knikure — APPROVED** [review]

**Pull request merged**